samcli/lib/providers/sam_api_provider.py

```diff
@@ -98,6 +115,100 @@ def _extract_from_serverless_api(
+                collector.set_default_authorizer(logical_id, auth.get(SamApiProvider._DEFAULT_AUTHORIZER))
+
+            self._extract_authorizers_from_props(logical_id, auth, collector, Route.API)
+
+    @staticmethod
+    def _extract_authorizers_from_props(logical_id: str, auth: dict, collector: ApiCollector, event_type: str) -> None:
+        """
+        Extracts Authorizers from the Auth properties section of Serverless resources
+
+        Parameters
+        ----------
+        logical_id: str
+            The logical ID of the Serverless resource
+        auth: dict
+            The Auth property dictionary
+        collector: ApiCollector
+            The Api Collector to send the Authorizers to
+        event_type: str
+            What kind of API this is (API, HTTP API)
+        """
+        API_IDENTITY_SOURCE_PREFIX = "method."
+        HTTP_IDENTITY_SOURCE_PREFIX = "$"
+
+        prefix = API_IDENTITY_SOURCE_PREFIX if event_type == Route.API else HTTP_IDENTITY_SOURCE_PREFIX
+        authorizers: Dict[str, Authorizer] = {}
+
+        for auth_name, auth_props in auth.get(SamApiProvider._AUTHORIZERS, {}).items():
+            authorizer_type = auth_props.get(SamApiProvider._FUNCTION_TYPE, LambdaAuthorizer.TOKEN).lower()
+            identity_object = auth_props.get(SamApiProvider._IDENTITY, {})
+
+            function_arn = auth_props.get(SamApiProvider._FUNCTION_ARN)
+
+            if not function_arn:
+                LOG.info("Authorizer '%s' is currently unsupported (must be a Lambda Authorizer), skipping", auth_name)
+                continue
+
+            function_name = LambdaUri.get_function_name(function_arn)
+
+            if not function_name:
+                LOG.warning("Unable to parse the Lambda ARN for Authorizer '%s', skipping", auth_name)
+                continue
+
+            if authorizer_type == LambdaAuthorizer.REQUEST or event_type == Route.HTTP:
+                identity_sources = []
+
+                for query_string in identity_object.get(SamApiProvider._IDENTITY_QUERY, []):
+                    identity_sources.append(f"{prefix}request.querystring.{query_string}")
+
+                for header in identity_object.get(SamApiProvider._IDENTITY_HEADERS, []):
+                    identity_sources.append(f"{prefix}request.header.{header}")
+
+                # context and stageVariables do not have "method." for V1 APIGW
+                # but the V2 still expects "$"
+                prefix = HTTP_IDENTITY_SOURCE_PREFIX if event_type == Route.HTTP else ""
+
+                for context in identity_object.get(SamApiProvider._IDENTITY_CONTEXT, []):
+                    identity_sources.append(f"{prefix}context.{context}")
+
+                for stage_variable in identity_object.get(SamApiProvider._IDENTITY_STAGE, []):
+                    identity_sources.append(f"{prefix}stageVariables.{stage_variable}")
+
+                payload_version = identity_object.get(SamApiProvider._AUTHORIZER_PAYLOAD, "1.0")
```

Comment:
nit: I saw V1 was added as constant in this or previous PR, can we use that?

Reply:
Yup, the constant was added in a previous PR but I can copy that over here/wait for the previous one to go in.